src/shogun/machine/gp/EPInferenceMethod.cpp

```diff
@@ -179,7 +181,7 @@ void CEPInferenceMethod::update()
 
 	// get and scale diagonal of the kernel matrix
 	SGVector<float64_t> ktrtr_diag=m_ktrtr.get_diagonal_vector();
-	ktrtr_diag.scale(CMath::exp(m_log_scale*2.0));
+	scale<linalg::Backend::NATIVE>(ktrtr_diag, CMath::exp(m_log_scale*2.0));
```

Comment:
you mean this overload has to be like in `SGVector::operator+=` (scale() method will be called inside the operator then)?